lib/galaxy/tools/__init__.py

```diff
@@ -3113,12 +3114,12 @@ def produce_outputs(self, trans, out_data, output_collections, incoming, history
 
         for dce in sorted_elements:
             dce_object = dce.element_object
-            copied_dataset = dce_object.copy(flush=False)
+            copied_dataset = dce_object.copy(copy_tags=dce_object.tags, flush=False)
```

Comment:
I don't think this will work for list:paired, element_object should be a `DatasetCollection` in that case
```suggestion
            copied_dataset = dce_object.copy(copy_tags=getattr(dce_object, 'tags', None), flush=False)
```